domain/secretbackend/state/types.go

```diff
@@ -196,15 +196,19 @@ func (rows secretBackendRows) toSecretBackends() []*secretbackend.SecretBackend
 			currentBackend = &backend
 			result = append(result, currentBackend)
 		}
-		if row.ConfigName == "" || row.ConfigContent == "" {
+		if row.ConfigName == "" {
 			// No config for this row.
 			continue
 		}
-
+		decodedContent, err := decodeConfigValue(row.ConfigContent)
+		if err != nil {
+			logger.Warningf(ctx, "failed to decode config value %q: %v", row.ConfigName, err)
```

Comment:
Not being able to read config content would be a problem, but this indeed should never happen as we always encode on the way in - can we add a comment that this should be unexpected.